src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageManagerViewModel.cs

```diff
@@ -307,55 +306,70 @@ internal class RPackageManagerViewModel : BindableBase, IRPackageManagerViewMode
             try {
                 await _packageManager.UnloadPackageAsync(package.Name);
             } catch (RHostDisconnectedException) {
-                AddErrorMessage(string.Format(CultureInfo.CurrentCulture, Resources.PackageManager_CantUnloadPackageNoRSession, package.Name));
+                AddErrorMessage(Resources.PackageManager_CantUnloadPackageNoRSession.FormatCurrent(package.Name), ErrorMessageType.PackageOperations);
             } catch (RPackageManagerException ex) {
-                AddErrorMessage(ex.Message);
+                AddErrorMessage(ex.Message, ErrorMessageType.PackageOperations);
             }
 
             await ReloadLoadedPackagesAsync();
             await AfterLoadUnloadAsync(package, startingTab);
         }
 
         private void BeforeLoadUnload(IRPackageViewModel package) {
-            if (_selectedTab == SelectedTab.InstalledPackages || _selectedTab == SelectedTab.LoadedPackages) {
+            if (_selectedTab == Tab.InstalledPackages || _selectedTab == Tab.LoadedPackages) {
                 IsLoading = true;
             }
             package.IsChanging = true;
         }
 
-        private async Task AfterLoadUnloadAsync(IRPackageViewModel package, SelectedTab startingTab) {
+        private async Task AfterLoadUnloadAsync(IRPackageViewModel package, Tab startingTab) {
             await ReplaceItemsAsync(startingTab);
             package.IsChanging = false;
         }
 
         public void DismissErrorMessage() {
             _coreShell.AssertIsOnMainThread();
             if (HasMultipleErrors) {
-                FirstError = _errorMessages.Dequeue();
                 HasMultipleErrors = _errorMessages.Count > 0;
+                FirstError = _errorMessages[0].Message;
+                _firstErrorType = _errorMessages[0].Type;
+                _errorMessages.RemoveAt(0);
             } else {
                 FirstError = null;
+                _firstErrorType = ErrorMessageType.NoError;
                 HasMultipleErrors = false;
             }
         }
+        
+        private void DismissErrorMessages(ErrorMessageType messageType) {
+            _coreShell.AssertIsOnMainThread();
+            _errorMessages.RemoveWhere(e => e.Type == messageType);
+            HasMultipleErrors = _errorMessages.Count > 0;
+
+            if (_firstErrorType == messageType) {
+                DismissErrorMessage();
+            }
+        }
 
         public void DismissAllErrorMessages() {
             _coreShell.AssertIsOnMainThread();
```

Comment:
Some of them appear to be bound to the collection state such as HasMultipleErrors which can be readonly props